models/user.go

```diff
@@ -1179,3 +1179,15 @@ func UnfollowUser(userID, followID int64) (err error) {
 	}
 	return sess.Commit()
 }
+
+// GetStarredRepos returns the repos starred by a particular user
+func GetStarredRepos(userID int64) ([]*Repository, error) {
```

Comment:
This function should return only stars of public repos by default, unless the user is the logged in user.

See https://github.com/go-gitea/gitea/pull/181 as example